src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private int currentCapacity;
+    private int currentLoadFactor;
+    private Node<K, V>[] storage;
+
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        currentLoadFactor = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        storage = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeed();
+        int position = getPosition(key);
+        Node<K, V> oldNode = storage[position];
+        if (oldNode == null) {
+            storage[position] = new Node<>(key, value);
+            size++;
+        } else {
+            Node<K, V> nextNode = oldNode;
```

Comment:
do you really need one more variable? You can use `node`

Reply:
++